src/lib/transaction_snark/test/zkapps_examples/calls/calls.ml

```diff
@@ -0,0 +1,381 @@
+          ~fee_payer_hash:
+            (Parties.Call_forest.Digest.Party.create
+               (Party.of_fee_payer fee_payer) )
+      in
+      let sign_all ({ fee_payer; other_parties; memo } : Parties.t) : Parties.t
+          =
+        let fee_payer =
+          match fee_payer with
+          | { body = { public_key; _ }; _ }
+            when Public_key.Compressed.equal public_key pk_compressed ->
+              { fee_payer with
+                authorization =
+                  Schnorr.Chunked.sign sk
+                    (Random_oracle.Input.Chunked.field full_commitment)
+              }
+          | fee_payer ->
+              fee_payer
+        in
+        let other_parties =
+          Parties.Call_forest.map other_parties ~f:(function
+            | ({ body = { public_key; use_full_commitment; _ }
+               ; authorization = Signature _
+               } as party :
+                Party.t )
+              when Public_key.Compressed.equal public_key pk_compressed ->
+                let commitment =
+                  if use_full_commitment then full_commitment
+                  else transaction_commitment
+                in
+                { party with
+                  authorization =
+                    Signature
+                      (Schnorr.Chunked.sign sk
+                         (Random_oracle.Input.Chunked.field commitment) )
+                }
+            | party ->
+                party )
+        in
+        { fee_payer; other_parties; memo }
+      in
+      let parties : Parties.t =
+        sign_all { fee_payer; other_parties = parties; memo }
+      in
+      Ledger.with_ledger ~depth:ledger_depth ~f:(fun ledger ->
+          let account =
+            Account.create account_id
+              Currency.Balance.(
+                Option.value_exn (add_amount zero (Currency.Amount.of_int 500)))
+          in
+          let _, loc =
+            Ledger.get_or_create_account ledger account_id account
+            |> Or_error.ok_exn
+          in
+          Async.Thread_safe.block_on_async_exn (fun () ->
+              check_parties_with_merges_exn ?expected_failure ledger [ parties ] ) ;
+          Ledger.get ledger loc )
+
+    let%test_unit "Initialize and update nonrecursive" =
+      let increments =
+        Array.init 1 ~f:(fun _ -> Snark_params.Tick.Field.random ())
+      in
+      let expected_result =
+        Array.reduce_exn ~f:Snark_params.Tick.Field.add increments
+      in
+      let account =
+        []
+        |> Parties.Call_forest.cons_tree
+             (Update_state_party.party (Add increments.(0)))
+        |> Parties.Call_forest.cons_tree Initialize_party.party
+        |> Parties.Call_forest.cons Deploy_party.party
+        |> test_parties
+      in
+      let (first_state :: zkapp_state) =
+        (Option.value_exn (Option.value_exn account).zkapp).app_state
+      in
+      assert (Snark_params.Tick.Field.equal expected_result first_state) ;
+      Pickles_types.Vector.iter
```

Comment:
this is to show that the rest of the state has not been modified? a comment to that effect?